### ISSUE TITLE
Design the architecture for tokenization of raw usfm

### ISSUE THREAD

**Issue by katherinejensen00** [paranext/paranext-core#540]
**User Story**
As an extension developer, I want to access tokenized Scripture data so it can be manipulated in my extension using a parser I am comfortable with. 

**Description**
Design the architecture for tokenization of raw usfm in a way that allows people to choose their parser (e.g. proskomma vs paratext)

Document in words and architectural diagram how we are going from usfm files on disk to characters in an editor and back.

Note: Martin Hosken created a USFM parser in Rust (but Martin is on furlough)

Matt G might have an idea about how to package Rust

Link to current usfm api: https://github.com/paranext/paranext-core/blob/666f74f2b0046493efcac1940a1fc0080f2d3fb8/extensions/src/usfm-data-provider/index.d.ts#L120

**Comment by tjcouch-sil** [comment]
Design what services we're going to provide to help the shared editor get broken up architecturally

Decide what input format is to shared editor. Work out where we have a layer of abstraction
Drop in USFM parser from Martin, Paratext, Proskomma, etc
What does Martin have with this Rust parser? Start with John Wickberg since Martin is on furlow
Glenn wants to use USFM-3 standard implementation from Martin

Proskomma is a way of parsing USFM into some kind of JSON format. Rust parser does the same. Ideal for people to choose in a config file somewhere. Glenn wants to use Rust from Martin.

**How can we integrate other USFM/tokenized sources (Proskomma, Rust parser) with existing ParatextData.dll tools like Biblical Terms and Checks and such?**

**Comment by tjcouch-sil** [comment]
[Decisions on USFM Tokenization](https://docs.google.com/document/d/1P7_G8WCPDuNtKlPvMcAP4SadewWauJFgkapS9gcpu1I/edit#heading=h.vb87ifmapgdr)

We need to talk to Glenn to confirm the paths forward.

We decided to wait on an architectural diagram since we don't have concrete decisions yet.

**Comment by tjcouch-sil** [comment]
Let's follow up on this in another issue
